### ISSUE TITLE
Use ProcessHandle API to check termination of parent process

### ISSUE THREAD

**Issue by rgrunber** [eclipse/lemminx#1217]
https://github.com/eclipse/lemminx/blob/master/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/commons/ParentProcessWatcher.java

Usually, the language server will be shut down by the client as defined by the [`shutdown()`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#shutdown) request of the LSP spec. In some cases this might fail, so we have the parent process watcher to handle it. On initialization, the client sends its process ID to the server and that class is used to monitor it (`kill -0` verifies the process is up).

It might be better to use the ProcessHandle API as opposed to spawning a separate process (eg. `kill`). I don't think we need to remove the existing code just yet. But maybe we can try ProcessHandle first, and only fall back to the old behaviour if that fails.

**Comment by datho7561** [comment]
This looks like it'd be really elegant, but according to the POM we're still using Java 8 for lemminx, so I don't think this is available unfortunately.

**Comment by rgrunber** [comment]
Right, newly introduced in Java 9. Given that getting people to have Java on their system, we probably don't want to force 11 as the minimum just yet. Although when lsp4j migrates to Java 11 we may get forced into it and have no choice but to also do this.
